flybywiresim/aircraft
# feat(fms): implement TOO STEEP PATH

Fixes #[issue_no]

## Summary of Changes

When the FMS predicts that a segment between two altitude constraints cannot be met, it shows a TOO STEEP PATH marker.
When the aircraft sequences the first of those constraints, the linear deviation (yoyo) will jump, putting the aircraft high on profile. When a vertical profile is predicted to have a TOO STEEP PATH segment, the "TOO STEEP PATH AHEAD" scratchpad message is shown. On the vertical revision page of the legs before and after the marker, the TOO STEEP PATH BEYOND indication is shown.

## Screenshots (if necessary)


![image](https://github.com/user-attachments/assets/38bc2741-ef88-4592-b29f-8a84c03dcc60)
![image](https://github.com/user-attachments/assets/0f85778c-4474-4d98-824c-c4648e193d50)


## References





## Additional context



Discord username (if different from GitHub):

## Testing instructions

This affects both the A32NX and the A380X.

- Spawn, enter a flight plan, ZFW, and cruise altitude to ensure you get vertical predictions in the flight plan.
- Enter an arrival/approach segment with a steep descent imposed by the constraints, check if the TOO STEEP PATH marker is shown. If you cannot find such a procedure, enter a really high constraint manually.
- Check the TOO STEEP PATH BEYOND message appears on the VERT REV page of the leg before and after the marker.
- Check the TOO STEEP PATH AHEAD scratchpad message appears in cruise when within 150 NM of the TOO STEEP PATH segment, or in the descent.


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page
